note/note_rfc6962_test.go

```diff
@@ -0,0 +1,168 @@
+	romeURL   = "https://rome.ct.filippo.io/2024h1/"
+	romePKDER = "MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEAXM8Ld9qn64g1zVFDh5FtgxS3zj5sqQDwYMs3wrBV3MCBiFhK/iRLxdKF4YsAcJaEglMlu4Lewvzxs0xO2uwEw=="
+)
+
+func TestRFC6962VerifierString(t *testing.T) {
+	for _, test := range []struct {
+		name    string
+		url     string
+		pubK    []byte
+		want    string
+		wantErr bool
+	}{
+		{
+			name: "works - argon ECDSA",
+			url:  "https://ct.googleapis.com/logs/us1/argon2024/",
+			pubK: mustB64(t, "MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEHblsqctplMVc5ramA7vSuNxUQxcomQwGAVAdnWTAWUYr3MgDHQW0LagJ95lB7QT75Ve6JgT2EVLOFGU7L3YrwA=="),
+			want: "ct.googleapis.com/logs/us1/argon2024+7deb49d0+BTBZMBMGByqGSM49AgEGCCqGSM49AwEHA0IABB25bKnLaZTFXOa2pgO70rjcVEMXKJkMBgFQHZ1kwFlGK9zIAx0FtC2oCfeZQe0E++VXuiYE9hFSzhRlOy92K8A=",
+		}, {
+			name: "works - rome ECDSA",
+			url:  romeURL,
+			pubK: mustB64(t, romePKDER),
+			want: "rome.ct.filippo.io/2024h1+78f4abae+BTBZMBMGByqGSM49AgEGCCqGSM49AwEHA0IABAFzPC3fap+uINc1RQ4eRbYMUt84+bKkA8GDLN8KwVdzAgYhYSv4kS8XSheGLAHCWhIJTJbuC3sL88bNMTtrsBM=",
+		}, {
+			name: "works - no scheme",
+			url:  "ct.googleapis.com/logs/us1/argon2024/",
+			pubK: mustB64(t, "MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEHblsqctplMVc5ramA7vSuNxUQxcomQwGAVAdnWTAWUYr3MgDHQW0LagJ95lB7QT75Ve6JgT2EVLOFGU7L3YrwA=="),
+			want: "ct.googleapis.com/logs/us1/argon2024+7deb49d0+BTBZMBMGByqGSM49AgEGCCqGSM49AwEHA0IABB25bKnLaZTFXOa2pgO70rjcVEMXKJkMBgFQHZ1kwFlGK9zIAx0FtC2oCfeZQe0E++VXuiYE9hFSzhRlOy92K8A=",
+		}, {
+			name: "works - no trailing slash",
+			url:  "ct.googleapis.com/logs/us1/argon2024",
+			pubK: mustB64(t, "MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEHblsqctplMVc5ramA7vSuNxUQxcomQwGAVAdnWTAWUYr3MgDHQW0LagJ95lB7QT75Ve6JgT2EVLOFGU7L3YrwA=="),
+			want: "ct.googleapis.com/logs/us1/argon2024+7deb49d0+BTBZMBMGByqGSM49AgEGCCqGSM49AwEHA0IABB25bKnLaZTFXOa2pgO70rjcVEMXKJkMBgFQHZ1kwFlGK9zIAx0FtC2oCfeZQe0E++VXuiYE9hFSzhRlOy92K8A=",
+		}, {
+			name:    "invalid name",
+			url:     "ct.googleapis.com/logs/us1/argon2024+cheese",
+			pubK:    mustB64(t, "MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEHblsqctplMVc5ramA7vSuNxUQxcomQwGAVAdnWTAWUYr3MgDHQW0LagJ95lB7QT75Ve6JgT2EVLOFGU7L3YrwA=="),
+			wantErr: true,
+		},
+	} {
+		t.Run(test.name, func(t *testing.T) {
+			k, err := x509.ParsePKIXPublicKey(test.pubK)
+			if err != nil {
+				t.Fatalf("Bad test data, couldn't parse key: %v", err)
+			}
+			got, err := RFC6962VerifierString(test.url, k)
+			if gotErr := err != nil; gotErr != test.wantErr {
+				t.Fatalf("Got err %v, but wantErr: %T", err, test.wantErr)
+			}
+			if got != test.want {
+				t.Fatalf("Got %q, want %q", got, test.want)
+			}
+		})
+	}
+}
+
+func TestVerify(t *testing.T) {
+	for _, test := range []struct {
+		name     string
+		cp       []byte
+		verifier string
+		wantErr  bool
```

Comment:
Will you use this in a followup PR? tests are complaining about untested paths. But also.. have fun with bits.

Reply:
Good catch, test added - thanks!